oauth2/src/main/java/com/linecorp/armeria/common/auth/oauth2/OAuth2TokenDescriptor.java

```diff
@@ -0,0 +1,511 @@
+    public String userName() {
+        return userName;
+    }
+
+    /**
+     * {@code token_type} Token Introspection Response field,
+     * OPTIONAL. Type of the token as defined at
+     * <a href="http://tools.ietf.org/html/rfc6749#section-7.1">[RFC6749], Section 7.1</a>.
+     */
+    @Nullable
+    public String tokenType() {
+        return tokenType;
+    }
+
+    /**
+     * {@code exp} Token Introspection Response field,
+     * OPTIONAL. {@link Instant} timestamp, indicating when this token will expire, as defined at
+     * <a href="https://tools.ietf.org/html/rfc7519">[RFC7519]</a>.
+     */
+    @Nullable
+    public Instant expiresAt() {
+        return expiresAt;
+    }
+
+    /**
+     * {@code iat} Token Introspection Response field,
+     * OPTIONAL. {@link Instant} timestamp, indicating when this token was originally issued, as defined at
+     * <a href="https://tools.ietf.org/html/rfc7519">[RFC7519]</a>.
+     */
+    @Nullable
+    public Instant issuedAt() {
+        return issuedAt;
+    }
+
+    /**
+     * {@link Duration} indicating the lifetime of the access token. The value is NOT supplied with the
+     * Token Introspection response and calculated based on {@code iat} and {@code exp} response fields every
+     * time this method invoked.
+     */
+    @Nullable
+    public Duration expiresIn() {
+        if (issuedAt != null && expiresAt != null) {
+            return Duration.ofMillis(issuedAt.until(expiresAt, ChronoUnit.MILLIS));
+        }
+        return null;
+    }
+
+    /**
+     * {@code nbf} Token Introspection Response field,
+     * OPTIONAL. {@link Instant} timestamp, indicating when this token is not to be used before, as defined at
+     * <a href="https://tools.ietf.org/html/rfc7519">[RFC7519]</a>.
+     */
+    @Nullable
+    public Instant notBefore() {
+        return notBefore;
+    }
+
+    /**
+     * {@code sub} Token Introspection Response field,
+     * OPTIONAL. Subject of the token. Usually a machine-readable
+     * identifier of the resource owner who authorized this token. As defined at
+     * <a href="https://tools.ietf.org/html/rfc7519">[RFC7519]</a>.
+     */
+    @Nullable
+    public String subject() {
+        return subject;
+    }
+
+    /**
+     * {@code aud} Token Introspection Response field,
+     * OPTIONAL. Service-specific string identifier or list of string identifiers representing the
+     * intended audience for this token, as defined at
+     * <a href="https://tools.ietf.org/html/rfc7519">[RFC7519]</a>.
+     */
+    @Nullable
+    public String audience() {
+        return audience;
+    }
+
+    /**
+     * {@code iss} Token Introspection Response field,
+     * OPTIONAL. String representing the issuer of this token, as defined at
+     * <a href="https://tools.ietf.org/html/rfc7519">[RFC7519]</a>.
+     */
+    @Nullable
+    public String issuer() {
+        return issuer;
+    }
+
+    /**
+     * {@code jti} Token Introspection Response field,
```

Comment:
Could use `{@value ...}`

Reply:
Somehow it does not work well with static imports

Reply:
fixed